### PR TITLE
Update with ADR contents; rename AWS SSO to AWS IAM Identity Center

### LINKED ISSUE #99: AWS SSO renamed to AWS IAM Identity Center
Should go through code comments, documentation, examples, etc. for SSO Extensions and update it to account for the rename. I don't think any links or low level API changes occurred with the rename, and all the resources, in IAM, etc. continue to reference `sso:`, `identitystore:`, etc. so we shouldn't have to adjust any code.

I wonder if this is heralding an incoming CRUD API other than SCIM for users... :)

### PR DESCRIPTION
**Pull request by leelalagudu**
*Issue #, if available:* Fixes #99

*Description of changes:*

- Added ADR section to reflect design choices
- Renamed all instances of AWS SSO with AWS IAM Identity Center

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,29 +25,29 @@
...
     "@types/js-yaml": "^4.0.5",
-    "@types/node": "^18.7.13",
+    "@types/node": "^18.7.15",
     "@types/uuid": "^8.3.4",
-    "@typescript-eslint/eslint-plugin": "^5.35.1",
-    "@typescript-eslint/parser": "^5.35.1",
-    "aws-cdk": "^2.39.0",
-    "esbuild": "^0.15.5",
+    "@typescript-eslint/eslint-plugin": "^5.36.2",
+    "@typescript-eslint/parser": "^5.36.2",
```
[the remark this answers is not in the document]
Comment (leelalagudu) [reply package.json:34]: This is an upkeep operation where I upgrade the packages on every PR, no specific reasons as such

[the remark this answers is not in the document]
Comment (tamara-h) [reply package.json:34]: As the automated checks are not running, and this is a documentation PR can this be left out? thank you

[the remark this answers is not in the document]
Comment (leelalagudu) [reply package.json]: Reverted the package version changes back

**Comment by jmejco** [comment]
@leelalagudu Can you please close and re-open this one to trigger the checks? I have updated the source.

**Pull request closed without merging**